### PR TITLE
VideoPress: Add chapters validation function

### PR DESCRIPTION
**Pull request by dhasilva**
Part of #28588

## Proposed changes:

* Adds a chapters validation function, but does not use it yet.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Locally, `jetpack test packages/videopress js`
* Go to the block editor and add a VideoPress block
* Add some chapters in the description
* Check that the functionality still works as normal

### CODE REVIEW

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	return (
+		parseInt( timeSections[ 0 ] ) * 3600 +
+		parseInt( timeSections[ 1 ] ) * 60 +
+		parseInt( timeSections[ 2 ] )
+	);
+}
+
+/**
+ *	Validates the generated chapters against UX restrictions
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts]:
```suggestion
 * Validates the generated chapters against UX restrictions
```

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * @param {VideoPressChapter[]} chapters - The chapters to be validated
+ * @returns {boolean} - Whether the generated chapters are valid or not
+ */
+export default function validateChapters( chapters: VideoPressChapter[] ) {
+	if ( ! chapters || chapters.length === 0 ) {
+		return false;
+	}
+
+	// The first timestamp should be 00:00:00
+	if ( chapters[ 0 ].startAt !== '00:00:00' ) {
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts:31]: The documentation states:

> Make sure that the first timestamp starts with 00:00.

So I think we should allow the `mm:ss` format, as well as the `hh:mm:ss`, maybe this way:

```suggestion
	if ( chapters[ 0 ].startAt !== '00:00' && chapters[ 0 ].startAt !== '00:00:00' ) {
```

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import type { VideoPressChapter } from '../types';
+
+/**
+ * Calculates the number of seconds from a timestamp in the format hh:mm:ss
+ *
+ * @param {string} time - The timestamp
+ * @returns {number}    - The number of seconds
+ */
+function getSeconds( time: string ): number {
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts:9]: We probably need to change this to be able to support the `mm:ss` format (see comment below).

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts (hunk 2 above)
Comment (lhkowalski) [reply projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts:31]: Never mind. I was checking the extraction logic and it does the magic for us, transforming any valid-ish format of `hh:mm:ss` or `mm:ss` or `m:ss` to the right `hh:mm:ss` format so here we are already dealing with clean values 🏆

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts (hunk 3 above)
Comment (lhkowalski) [reply projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts:9]: Never mind, comment below explains why this is not necessary.

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/test/index.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { validateChapters } from '..';
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/test/index.ts]: This tests are 💯

**Review by lhkowalski — DISMISSED** [review]
Excelent work. Left some comments but I was discovering the things about this features as I reviewed, so in the end the comments were not necessary.

Extra kudos for moving things around to help the organization.

File: projects/packages/videopress/src/client/block-editor/hooks/use-video-data-update/index.ts [hunk 5]
```diff
@@ -19,9 +19,9 @@ import {
 	WPComV2VideopressGetMetaEndpointResponseProps,
 	WPComV2VideopressPostMetaEndpointBodyProps,
 } from '../../../types';
-import extractVideoChapters from '../../../utils/extract-video-chapters';
-import generateChaptersFile from '../../../utils/generate-chapters-file';
 import { snakeToCamel } from '../../../utils/map-object-keys-to-camel-case';
+import extractVideoChapters from '../../../utils/video-chapters/extract-video-chapters';
+import generateChaptersFile from '../../../utils/video-chapters/generate-chapters-file';
```
Comment (retrofox) [inline projects/packages/videopress/src/client/block-editor/hooks/use-video-data-update/index.ts:24]: <3

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	);
+}
+
+/**
+ * Validates the generated chapters against UX restrictions
+ *
+ * @param {VideoPressChapter[]} chapters - The chapters to be validated
+ * @returns {boolean} - Whether the generated chapters are valid or not
+ */
+export default function validateChapters( chapters: VideoPressChapter[] ) {
```
Comment (retrofox) [inline projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts]:
```suggestion
export default function validateChapters( chapters: VideoPressChapter[] ): boolean {
```

won't hurt

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+		if ( getSeconds( nextChapter.startAt ) - getSeconds( currentChapter.startAt ) < 10 ) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+export { validateChapters };
```
Comment (retrofox) [inline projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts]: We are exposing this function as default. Not need to do it here.

Comment (retrofox) [reply projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/index.ts]:
```suggestion
```

File: projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/test/index.ts (hunk 4 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/utils/video-chapters/validate-chapters/test/index.ts]:
```suggestion
import validateChapters from '..';
```

**Review by retrofox — DISMISSED** [review]
LGTM. I left a few minor comments. Exporting the same function twice maybe should be avoided. Anyway... 🚀

**Review by lhkowalski — APPROVED** [review]
Nice improvements incorporating @retrofox comments 🚀

**Review by retrofox — APPROVED** [review]

**Pull request merged**